src/components/LeafletMap.js

```diff
@@ -131,26 +130,42 @@ class LeafletMap extends Component {
     }
   }
 
+  setIconActive(marker) {
+    marker.setIcon(activeIcon);
+    this.setState({ highlightMarker: marker });
+  }
+
+  panToOffCenter(latlng, offset, options) {
+    const x = latlng.containerPoint.x + offset[0],
+      y = latlng.containerPoint.y - offset[1],
+      currentZoom = this.map._zoom,
+      newLatlng = this.map.containerPointToLatLng([x, y]);
+
+    return this.map.setView(newLatlng, currentZoom, { pan: options });
+  }
+
   updateMarkers(markersData) {
     const markersCluster = L.markerClusterGroup({
       showCoverageOnHover: false,
       spiderfyOnMaxZoom: false
     });
 
-    this.layer.clearLayers();
+    markersCluster.on('clusterclick', () => {
+      this.props.fullMapZoomHandler(false);
+    });
+
     markersData.forEach(marker => {
       const markerPoint = L.marker(
         [marker.geometry.coordinates[1], marker.geometry.coordinates[0]],
         { icon }
       );
 
       markersCluster.addLayer(
-        markerPoint.on('click', () => {
+        markerPoint.on('click', e => {
           this.removeHighlight();
-          markerPoint.setIcon(activeIcon);
-          this.setState({ highlightMarker: markerPoint });
+          this.setIconActive(markerPoint);
           this.props.siteClickHandler(marker);
-          this.map.flyTo([marker.geometry.coordinates[1], marker.geometry.coordinates[0]]);
+          this.panToOffCenter(e, [350, 0], { animate: true });
```

Comment:
The number `350`. How do you come to it, and do you know if it will be the correct number in all of the cases?